### ISSUE TITLE
OrderedSet is changing item ordering (@4.0.0-rc.11+)

### ISSUE THREAD

**Issue by pixmaster** [immutable-js/immutable-js#1663]
```javascript
const { OrderedSet, Map } = require('immutable@4.0.0-rc.11')
const i=[2];
OrderedSet([ Map({b:1, a:true}), Map({b:2,a:true}), Map({b:3,a:true}) ])
  .map((t) => {
  if (i.includes(t.get('b'))) return t.set('a', false);
  return t;
  })
  .toJS()
```
Result
![image](https://user-images.githubusercontent.com/3305295/49880299-8355e880-fe34-11e8-9d5a-eb0a90438094.png)

**Comment by Soreine** [comment]
Your example looks like the expected behavior.
During the `map`, if you return a different value, then it will probably `remove` the old value and `add`  the new value, effectively pushing it to the end of the new `OrderedSet`

**Comment by pixmaster** [comment]
@Soreine it only looks like, because I expect than `OrderedSet` don't change order in any cases. What the sense have type is `OrderedSet` if that changes order?

**Comment by B-R-Bender** [comment]
same with me, I've expected order to be persist

**Comment by B-R-Bender** [comment]
my workarount is to converto to List first and after updates is done convert it back to OrderedSet

```
return lineData.toList().map(dataItem => dataItem.update(
                            Y_VALUE,
                            value => value + prevLineData.getIn([dataItem.get(X_VALUE), Y_VALUE])
                        )).toOrderedSet();
```

**Comment by jdeniau** [comment]
Thank you for your bug report. The immutable-js oss fork will soon be merged and a fix for this issue has been included in #1833. Once this PR is merged, this issue will be resolved in the main branch. We will then do our best to to release the `4.0.0` version.

Commit reference: https://github.com/immutable-js/immutable-js/pull/1833/commits/37c34ecdc7098e70614e420f5f5f945beff46052

**Comment by jdeniau** [comment]
Hi!
The 4.0.0-rc.14 version has been released and should fix this issue.
Feel free to test it and re-open this issue if the issue is still present.
Thank you!
